group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -299,10 +299,10 @@ public CompletableFuture<JoinGroupResponseData> joinGroup(
             topicPartitionFor(request.groupId()),
             coordinator -> coordinator.genericGroupJoin(context, request, responseFuture)
         ).exceptionally(exception -> {
-            log.error("Request {} hit an unexpected exception: {}",
-                request, exception.getMessage());
-
             if (!responseFuture.isDone()) {
+                log.error("Request {} hit an unexpected exception: {}",
```

Comment:
This would still log in expected cases, no? For instance, when the coordinator for the group is inactive, loading, etc. If you really want to log something, you could perhaps log only if `exception` is not a KafkaException or only when it is a RuntimeException for instance.

Reply:
ah makes sense. logging only when it is not a kafka exception makes sense.